### PR TITLE
feat: add initial project structure, tests and documentation

### LINKED ISSUE #109: AlgoKit Smart Contract Building Blocks library
Acceptance Criteria:

* There will be a GitHub repository that publishes a PyPi package (with automated CI/CD, release notes and versioning)
* There will be a structure where different arcs can be contributed as a Beaker base class, the audit report, any (Markdown) documentation
* The initial set of smart contracts will be determined in the future
* There will be a CONTRIBUTING.md that outlines that all implementations will be audited and will implement a finalised ARC standard, for now it will note that contributions will be from Algorand Foundation only, but if someone wants to email arcs@algorand.foundation.
* Community templates can be managed separately (potentially cloning this repository)

Original notes:

A library for Algorand exists with audited, documented smart contracts and base smart contract templates in PyTEAL that allows Algorand Foundation contributed smart contracts (e.g. ARC reference implementations) and community contributed smart contracts

Inspiration:

* https://github.com/transmissions11/solmate
* https://www.openzeppelin.com/

### PR DESCRIPTION
**Pull request by daniel-makerx**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+black = {extras = ["d"], version = "^22.10.0"}
+ruff = "^0.0.191"
+pip-audit = "^2.4.7"
+pytest-mock = "^3.10.0"
+mypy = "^0.991"
+python-semantic-release = "^7.32.2"
+pytest-cov = "^4.0.0"
+pre-commit = "^2.20.0"
+markdown = "^3.4.1"
+bs4 = "^0.0.1"
```
Comment (robdmoore) [inline pyproject.toml:24]: What is this?

File: tests/test_repo_conventions.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
+import re
+from pathlib import Path
+from typing import Iterable
+
+import algokit_arc
+import markdown
+import pytest
+from beaker.application import Application
+from bs4 import BeautifulSoup
```
Comment (robdmoore) [inline tests/test_repo_conventions.py:9]: Right. Might be worth using the proper package name (look at the pypi page for bs4)

**Review by robdmoore — APPROVED** [review]

File: pyproject.toml (hunk 1 above)
Comment (daniel-makerx) [reply pyproject.toml:24]: It is used in this test to inspect the documentation for some basic links https://github.com/algorandfoundation/algokit-arc-library/pull/1/files#diff-15dfeaba21e5e85d8deed84ee029e3d47fae46232e7b857d12ed8a66d9cb5bdeR16

File: tests/test_repo_conventions.py (hunk 2 above)
Comment (daniel-makerx) [reply tests/test_repo_conventions.py:9]: Ah thanks, didn't realise the package name was different than the module name

**Pull request merged**